src/FieldWarning/Assets/Units/Component/Movement/Pathfinder.cs

```diff
@@ -55,6 +57,10 @@ public class Pathfinder
         /// </summary>
         private static bool _s_straightStep;
 
+        private Semaphore _PathFinderRunnerSem = new Semaphore(0,1);
```

Comment:
Also, I find it really important to document what variables this is locking. Is it the entire class? ```PathFinderRunner``` is a method, we don't lock methods, we lock state.

Reply:
It is FindPath that is being synchronized.. The semaphore is just for signaling